packages/x-components/src/plugins/x-plugin.mixin.ts

```diff
@@ -60,11 +61,15 @@ export function getBusAPI(bus: XBus, component: PrivateExtendedVueComponent): XC
     emit: <Event extends XEvent>(
       event: Event,
       payload?: XEventPayload<Event>,
-      metadata: Omit<WireMetadata, 'moduleName'> = {}
+      metadata: Omit<WireMetadata, 'moduleName' | 'origin'> = {}
     ) => {
       const xComponent = component.xComponent;
-      const moduleName = xComponent ? getXComponentXModuleName(xComponent) : null;
-      bus.emit(event, payload as any, { moduleName, origin: component.$origin, ...metadata });
+      bus.emit(event, payload as any, {
+        moduleName: xComponent ? getXComponentXModuleName(xComponent) : null,
+        origin: createQueryOrigin(metadata.feature, component.$location as QueryLocation),
```

Comment:
I'm not sure this is the right place to do this. The event should contain the feature and the location. Then the search  
 merge them before send to the API. Once the API responses with the tagging object, that info is captured by Tagging Module, and tracked to Tagging API.
When a result click/addToCart/... is done, the origin of the query is already in the Tagging Module, to add to the click track.

Reply:
I think this is the right place, the components are the only place you can retrieve both the location and the feature, and merge them into an origin. I see this as "enriching the metadata". If not you would have to do this in a mutation.